go/service/cryptocurrency.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package service
+
+import (
+	"github.com/keybase/client/go/engine"
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+	"golang.org/x/net/context"
+)
+
+type CryptocurrencyHandler struct {
+	*BaseHandler
+	libkb.Contextified
+}
+
+func NewCryptocurrencyHandler(xp rpc.Transporter, g *libkb.GlobalContext) *CryptocurrencyHandler {
+	return &CryptocurrencyHandler{
+		BaseHandler:  NewBaseHandler(xp),
+		Contextified: libkb.NewContextified(g),
+	}
+}
+
+// BTC creates a BTCEngine and runs it.
+func (h *CryptocurrencyHandler) RegisterAddress(_ context.Context, arg keybase1.RegisterAddressArg) error {
+	ctx := engine.Context{
+		LogUI:     h.getLogUI(arg.SessionID),
+		SecretUI:  h.getSecretUI(arg.SessionID, h.G()),
+		SessionID: arg.SessionID,
```

Comment:
if you're here, it wouldn't hurt to add `NetContext: nctx`, and replace `_` in params with `nctx`.  Not needed obviously but